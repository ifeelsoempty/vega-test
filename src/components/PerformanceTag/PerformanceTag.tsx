import { FC } from "react"
import { Tag } from "antd"

import { formatPrice } from "utils/formatePrice/formatPrice"

type PerformanceTagProps = {
  percent: number,
  number: number,
}

const getColor = (price: number) => {
  if(price > 0) {
    return 'success';
  }

  if(price < 0) {
    return 'error';
  }

  return 'default'
}

export const PerformanceTag: FC<PerformanceTagProps> = ({ percent, number }) => {
  const color = getColor(number);
  const formattedNumber = formatPrice(number, { withSign: true });
  const formattedPercent = Math.abs(percent).toFixed(2);

  return (
    <Tag color={color}>
      {formattedNumber} Â· {formattedPercent}%
    </Tag>
  )
}
