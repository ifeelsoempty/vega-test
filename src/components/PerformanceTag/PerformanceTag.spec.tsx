import { render } from '@testing-library/react';
import { PerformanceTag } from './PerformanceTag';

describe('PerformanceTag', () => {
  it('renders correctly with positive number', () => {
    const { container } = render(<PerformanceTag percent={10} number={100} />);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with negative number', () => {
    const { container } = render(<PerformanceTag percent={-5} number={-50} />);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with zero number', () => {
    const { container } = render(<PerformanceTag percent={0} number={0} />);
    expect(container).toMatchSnapshot();
  });
});
