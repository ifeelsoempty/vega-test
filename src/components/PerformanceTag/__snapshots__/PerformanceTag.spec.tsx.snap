// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PerformanceTag renders correctly with negative number 1`] = `
<div>
  <span
    class="ant-tag ant-tag-error css-dev-only-do-not-override-1drr2mu"
  >
    -$50.00
     · 
    5.00
    %
  </span>
</div>
`;

exports[`PerformanceTag renders correctly with positive number 1`] = `
<div>
  <span
    class="ant-tag ant-tag-success css-dev-only-do-not-override-1drr2mu"
  >
    +$100.00
     · 
    10.00
    %
  </span>
</div>
`;

exports[`PerformanceTag renders correctly with zero number 1`] = `
<div>
  <span
    class="ant-tag ant-tag-default css-dev-only-do-not-override-1drr2mu"
  >
    $0.00
     · 
    0.00
    %
  </span>
</div>
`;
